keras_cv/utils/preprocessing.py

```diff
@@ -323,3 +324,51 @@ def check_fill_mode_and_interpolation(fill_mode, interpolation):
             "Unknown `interpolation` {}. Only `nearest` and "
             "`bilinear` are supported.".format(interpolation)
         )
+
+
+def clip_bounding_box(bounding_boxes, bounding_box_format, images):
+    """clips bounding boxes to image boundaries"""
+    boxes_rank = len(bounding_boxes.shape)
+    images_rank = len(images.shape)
+    boxes_includes_batch = boxes_rank == 3
+    images_include_batch = images_rank == 4
+    if boxes_includes_batch != images_include_batch:
+        raise ValueError(
+            "clip_bounding_box() expects both boxes and images to be batched, or both "
+            f"boxes and images to be unbatched.  Received len(boxes.shape)={boxes_rank}, "
+            f"len(images.shape)={images_rank}.  Expected either len(boxes.shape)=2 AND "
+            "len(images.shape)=3, or len(boxes.shape)=3 AND len(images.shape)=4."
+        )
+
+    def clip_unbatched_bbox(bbox):
+        x1, y1, x2, y2 = tf.split(bbox, 4, axis=1)
+        clipped_bboxes = tf.stack(
+            [
+                tf.clip_by_value(x1, clip_value_min=0, clip_value_max=1),
```

Comment:
do we need to solve this with a map_fn?  can we do it natively in a vectorized manner?

Reply:
Updated!